test/forms/InflationProofFormSpec.scala

```diff
@@ -70,87 +71,87 @@ class inflationProofFormSpec extends PlaySpec with GuiceOneAppPerSuite {
-        inflationProofFormResults.errors must contain(FormError("revaluationDate", List(Messages("gmp.error.date.nonnumber"))))
+        inflationProofFormResults.errors must contain(invalidDateError("revaluationDate.year"))
       }
 
       "return an error on the date when it is not the correct format" in {
         val inflationProofFormResults = inflationProofForm.bind(Json.toJson(InflationProof(new GmpDate(Some("01"), Some("11"), Some("190")),Some("Yes"))), fromJsonMaxChars)
-        inflationProofFormResults.errors must contain(FormError("revaluationDate", List(Messages("gmp.error.year.invalid.format"))))
-        inflationProofFormResults.errors must not contain(FormError("revaluationDate", List(Messages("gmp.error.year.invalid"))))
+        inflationProofFormResults.errors must contain(invalidDateError("revaluationDate.year", "error.yearLength"))
+        inflationProofFormResults.errors must not contain(invalidDateError("revaluationDate.day"))
       }
     }
 
     "entering invalid dates" must {
       "return an error when the day is missing" in {
         val inflationProofFormResults = inflationProofForm.bind(Json.toJson(InflationProof(new GmpDate(Some(""), Some("04"), Some("1978")),Some("Yes"))), fromJsonMaxChars)
-        inflationProofFormResults.errors must contain(FormError("", List(Messages("gmp.error.date.invalid"))))
+        inflationProofFormResults.errors must contain(invalidDateError("revaluationDate.day", "error.dayRequired"))
       }
       "return an error when the month is missing" in {
         val inflationProofFormResults = inflationProofForm.bind(Json.toJson(InflationProof(new GmpDate(Some("04"), Some(""), Some("1978")),Some("Yes"))), fromJsonMaxChars)
-        inflationProofFormResults.errors must contain(FormError("", List(Messages("gmp.error.date.invalid"))))
+        inflationProofFormResults.errors must contain(invalidDateError("revaluationDate.month", "error.monthRequired"))
       }
       "return an error when the year is missing" in {
         val inflationProofFormResults = inflationProofForm.bind(Json.toJson(InflationProof(new GmpDate(Some("04"), Some("04"), Some("")),Some("Yes"))), fromJsonMaxChars)
-        inflationProofFormResults.errors must contain(FormError("", List(Messages("gmp.error.date.invalid"))))
+        inflationProofFormResults.errors must contain(invalidDateError("revaluationDate.year", "error.yearRequired"))
       }
     }
 
     "entering a date outside valid GMP dates" must {
 
       "return an error when before 05/04/1978" in {
         val inflationProofFormResults = inflationProofForm.bind(Json.toJson(InflationProof(new GmpDate(Some("04"), Some("04"), Some("1978")),Some("Yes"))), fromJsonMaxChars)
-        inflationProofFormResults.errors must contain(FormError("", List(Messages("gmp.error.reval_date.from"))))
+        inflationProofFormResults.errors must contain(FormError("revaluationDate", List("error.tooFarInPast"), List("6 April 1978")))
       }
```

Comment:
this test is only passing because the wrong argument is being passed, when the arg is updated in the form mapping call then this should read `List("4 April 1978")`

Reply:
the test date should remain at "04" to be the same test  - it will throw the required error if it is "04" and that tests the inclusivity of the minimum date